Corona-Warn-App/src/main/java/de/rki/coronawarnapp/diagnosiskeys/storage/legacy/LegacyKeyCacheMigration.kt

```diff
@@ -0,0 +1,92 @@
+package de.rki.coronawarnapp.diagnosiskeys.storage.legacy
+
+import android.content.Context
+import dagger.Lazy
+import de.rki.coronawarnapp.util.HashExtensions.hashToMD5
+import de.rki.coronawarnapp.util.TimeStamper
+import kotlinx.coroutines.sync.Mutex
+import kotlinx.coroutines.sync.withLock
+import org.joda.time.Duration
+import org.joda.time.Instant
+import timber.log.Timber
+import java.io.File
+import javax.inject.Inject
+
+class LegacyKeyCacheMigration @Inject constructor(
+    private val context: Context,
+    private val legacyDao: Lazy<KeyCacheLegacyDao>,
+    private val timeStamper: TimeStamper
+) {
+
+    private val cacheDir by lazy {
+        File(context.cacheDir, "key-export")
+    }
+
+    private val workMutex = Mutex()
+    private var isInit = false
+    private val legacyCacheMap = mutableMapOf<String, File>()
+
+    private suspend fun tryInit() {
+        if (isInit) return
```

Comment:
Yes, but that would be okay, the init guard here is just to prevent (costly) checking on each call from `KeyFileDownloader`. We do it once on first call, per app session.

I've made 2 additional tweaks:
* A `if(!cacheDir.exists()) return` right after init, because I'd rather do an IO check, than the `legacyDB.clear()` on each app init due to our woes with SQL.
* And a check after init that deletes the cache dir if it exists and is empty, so the quick abort actually works.